sqlglot/optimizer/simplify.py

```diff
@@ -854,15 +846,27 @@ def cast_value(value: t.Any, to: exp.DataType) -> t.Optional[t.Union[datetime.da
     return None
 
 
-def extract_date(cast: exp.Cast) -> t.Optional[t.Union[datetime.date, datetime.date]]:
+def extract_date(cast: exp.Expression) -> t.Optional[t.Union[datetime.date, datetime.date]]:
+    to: exp.DataType
```

Comment:
Is this needed? We can also move it inline in one of the following assignments

Reply:
whoops, you're right